src/main.rs

```diff
@@ -177,11 +208,36 @@ impl FromStr for Blake3Cid {
     }
 }
 
+async fn make_rpc_client(
+    rpc_port: Option<u16>,
+) -> anyhow::Result<RpcClient<ProviderService, QuinnConnection<ProviderResponse, ProviderRequest>>>
+{
+    let rpc_port = rpc_port.unwrap_or(RPC_PORT);
+    let endpoint = iroh::get::make_client_endpoint(None, vec![RPC_ALPN.to_vec()], false, false)?;
+    let addr: SocketAddr = SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::LOCALHOST, rpc_port));
+    let server_name = "localhost".to_string();
+    let connection = QuinnConnection::new(endpoint, addr, server_name);
+    let client = RpcClient::<ProviderService, _>::new(connection);
+    // Do a version request to check if the server is running.
+    let _version = tokio::time::timeout(Duration::from_secs(1), client.rpc(VersionRequest))
+        .await
+        .context("iroh server is not running")??;
+    Ok(client)
+}
+
 const PROGRESS_STYLE: &str =
     "{msg}\n{spinner:.green} [{elapsed_precise}] [{wide_bar:.cyan/blue}] {bytes}/{total_bytes} ({eta})";
 
-#[tokio::main(flavor = "multi_thread")]
-async fn main() -> Result<()> {
+fn main() -> Result<()> {
+    let rt = tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .build()?;
+    rt.block_on(main_impl())?;
+    rt.shutdown_timeout(Duration::from_millis(500));
```

Comment:
why is this construction needed?

Reply:
Added a comment. Without this tokio::io::copy(stdin(), ...) hangs forever unless you press enter...